common/lib/util/eventemitter.js

```diff
@@ -57,7 +57,7 @@ var EventEmitter = (function() {
 			/* ... or we take event to be the actual event name and listener to be all */
 		}
 		var listeners, idx = -1;
-		if(event === null) {
+		if((event === null) || (event === undefined)) {
```

Comment:
Well we also have to worry about `false`, `0`, `''`. Either we have an API that detects the types of all arguments and rejects invalid arguments, or, it behaves on the assumption that it is given a string, or we have something in-between as now.

I think the logic should really be:

- decide which overloaded signature is being used: for this check if `event === undefined`
- assume that the `event` argument, if provided, is a string (or coerce to a string) and then reject any invalid string values.

I don't think `event === null` has a right to special handling any more than `typeof(event) == 'number'`.